awx/main/management/commands/run_dispatcher.py

```diff
@@ -35,48 +35,35 @@ def add_arguments(self, parser):
                                   'running jobs will run to completion first'))
 
     def beat(self):
-        from celery import Celery
-        from celery.beat import PersistentScheduler
-        from celery.apps import beat
+        from awx.main.beat import Beat
 
-        class AWXScheduler(PersistentScheduler):
+        class AWXScheduler(Beat):
 
             def __init__(self, *args, **kwargs):
                 self.ppid = os.getppid()
                 super(AWXScheduler, self).__init__(*args, **kwargs)
 
-            def setup_schedule(self):
-                super(AWXScheduler, self).setup_schedule()
-                self.update_from_dict(settings.CELERYBEAT_SCHEDULE)
-
             def tick(self, *args, **kwargs):
                 if os.getppid() != self.ppid:
                     # if the parent PID changes, this process has been orphaned
                     # via e.g., segfault or sigkill, we should exit too
                     raise SystemExit()
                 return super(AWXScheduler, self).tick(*args, **kwargs)
 
-            def apply_async(self, entry, producer=None, advance=True, **kwargs):
+            def apply_async(self, entry, producer=None, advance=True, expiration=None, **kwargs):
                 for conn in connections.all():
```

Comment:
Seems like this code could just go in the base implementation?  This function signature is unique to celery.